### DISCUSSION TITLE
Transform session data item

### DISCUSSION THREAD

**Discussion by emdfonseca** [keystonejs/keystone#5412] [Feature requests]
In keystone-next (6) there is a method to setup the session item: `withItemData`. This implementation is quite handy because it lets you set the session data based on the results of a graphql query. However, there are situation where the resulting object could be changed, for example, you might want to normalise the resulting data.

Is there any way to perform this operation? If it does not yet exist I am willing to submit a PR adding an optional transformer function parameter to the withItemData method.

**Use case**
Your model consists of users, companies, and members. Members have different roles on each company. This relation leads to the existence of an associative table with extra relationship metadata.

When building the session data I want to query all roles and store them in the session as normalised data. Example: {name: "John Doe", roles: { "companyOneId": "owner", "companyTwoId": "admin"}}.

**Comment by bladey** [comment]
Thanks for the suggestion @emdfonseca!
